src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -194,10 +194,10 @@ export function useDiscoverState({
    */
   useEffect(() => {
     const unsubscribe = appStateContainer.subscribe(async (nextState) => {
-      const { hideChart, interval, sort, index } = state;
-      // chart was hidden, now it should be displayed, so data is needed
-      const chartDisplayChanged = nextState.hideChart !== hideChart && hideChart;
+      const { hideChart, interval, breakdownField, sort, index } = state;
+      const chartDisplayChanged = nextState.hideChart !== hideChart;
```

Comment:
We used to only refetch when the chart was shown, but now we're refetching when it's hidden too in order to keep Discover fetches in sync with unified histogram. Once we support manually refetching in unified histogram, we could revert to the old way if we wanted to.